contracts/lib/AmountDeriver.sol

```diff
@@ -35,16 +34,24 @@ contract AmountDeriver is AmountDerivationErrors {
     function _locateCurrentAmount(
         uint256 startAmount,
         uint256 endAmount,
-        uint256 elapsed,
-        uint256 remaining,
-        uint256 duration,
+        uint256 startTime,
+        uint256 endTime,
         bool roundUp
-    ) internal pure returns (uint256) {
+    ) internal view returns (uint256) {
         // Only modify end amount if it doesn't already equal start amount.
         if (startAmount != endAmount) {
             // Leave extra amount to add for rounding at zero (i.e. round down).
             uint256 extraCeiling = 0;
 
+            // Derive the duration for the order and place it on the stack.
+            uint256 duration = endTime - startTime;
+
+            // Derive time elapsed since the order started & place on stack.
+            uint256 elapsed = block.timestamp - startTime;
```

Comment:
this ended up making the change much bigger than expected, since changing the function from `pure` to `view` and relying on current timestamp force me to switch around a lot of code in the test suite

a better compromise could have been to instead keep the function `pure` and send in an extra `currentTime`. Someone please let me know if that's preferable, and I'll make the change

Reply:
I think deferring these calculations until necessary makes a lot of sense. If you add a section to the natspec here documenting the conditions asserted in `_verifyTime` you could also make this computation unchecked